.github/workflows/linting.yml

```
@@ -1,40 +1,22 @@
 name: Linting
 on:
+  push:
+    branches:
+      - master
   pull_request:
-    paths:
-      - '**/*.yaml'
-      - '**/*.yml'
   workflow_dispatch:
+
 concurrency:
   group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
   cancel-in-progress: true
+
 jobs:
   lint-ruby:
-    name: Ruby - ${{ matrix.os }} ${{ matrix.ruby }}
-    runs-on: ${{ matrix.os }}-latest
-    strategy:
-      fail-fast: false
-      matrix:
-        os: [ubuntu, windows]
-        ruby: ['2.7', '3.0', '3.1', '3.2', '3.3', 'head']
-        include:
-          - os: ubuntu
-            ruby: jruby-9.4
-          - os: windows
-            ruby: mingw
-        exclude:
-          - os: windows
-            ruby: head
     steps:
-      - name: Windows Specific
-        if: matrix.os == 'windows'
-        run: |
-          # Work around `Layout/EndOfLine: Carriage return character detected`
-          git config --system core.autocrlf false
       - uses: actions/checkout@v4
       - uses: ruby/setup-ruby@v1
         with:
-          ruby-version: ${{ matrix.ruby }}
+          ruby-version: '3.3'
```

Can you add a comment to make the intention clearer?

```suggestion
          ruby-version: '3.3' # Use the latest Ruby version.
```

---

Or maybe this can be declared as some variable (e.g. `latest_ruby_version`)

---

https://github.com/rubocop/rubocop/pull/12964#discussion_r1625852504

> engine only like ruby and truffleruby, uses the latest stable release of that implementation

`ruby-version: ruby`

So, no need to change this every year. Quite nice I think